### PR TITLE
[WIP] Use workflow dispatch that now works with GITHUB_TOKEN

### PR DESCRIPTION
**Pull request by samueljseay**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

As of September workflow dispatch can now be done with `GITHUB_TOKEN`. See [this blog post](https://github.blog/changelog/2022-09-08-github-actions-use-github_token-with-workflow_dispatch-and-repository_dispatch/) for more info.

This allows us to remove one of the few remaining custom tokens we were using to get around those limitations.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1) Fork this repository.
2) Enable issues in your fork from `https://github.com/YOURORG/woocommerce/settings`
3) Enable actions in your fork from `https://github.com/YOURORG/woocommerce/settings/actions`. Also, make sure workflows have read and write permissions and the ability to create and approve PRs.
4) Enable the scheduled workflow from `https://github.com/YOURORG/woocommerce/actions/workflows/release-code-freeze.yml`.
6) Merge this PR into `trunk` of your new fork.
7) In your fork, create a release `7.3.0`.
8) If there aren't already changes in `plugins/woocommerce/changelog`, in trunk, create several changes in `trunk` that add change files to the changelog dir (whether via PRs or just manually adding change files).
9) Run the action manually from `https://github.com/YOURORG/woocommerce/actions/workflows/release-code-freeze.yml` with "2022-07-14" as the value for time override. You may disable the slack ping or change the channel if you want.
10) The script should run, create `release/7.4` from trunk, create a `7.4.0` milestone, generate the changelog, create a PR against `trunk` that removes the change files, and a PR against `release/7.3` that updates the changelog and removes the change files.
11) **If you need to repeat testing, first delete the branches and the milestone created by this PR.**
12) **Remember, you've enabled the action, which should also run according to the cron schedule.** If you don't want that to happen, disable the action again before it runs. (It should be relatively harmless in your fork)

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: .github/workflows/release-code-freeze.yml [hunk 1]
```diff
@@ -21,142 +21,145 @@ env:
...
-  trigger-changelog-action:
-    name: "Trigger changelog action"
-    runs-on: ubuntu-20.04
-    needs: maybe-create-next-milestone-and-release-branch
-    steps:
-      - run: |
-          curl \
-            -X POST \
-            -H "Accept: application/vnd.github.v3+json" \
-            -H "Authorization: token ${{ secrets.WC_BOT_TRIAGE_TOKEN }}" \
```
Comment (samueljseay) [inline .github/workflows/release-code-freeze.yml:160]: Its not obvious in the changeset due to prettier reformatting files, but this is the token we're trying to get rid of.

**Comment by samueljseay** [comment]
@jonathansadowski just wondering if I could get some assistance and ideas around testing this.

I assume I'll have to go through code freeze process in a fork or something?

**Comment by jonathansadowski** [comment]
> @jonathansadowski just wondering if I could get some assistance and ideas around testing this.
> 
> I assume I'll have to go through code freeze process in a fork or something?

Yep, I'd probably follow a process pretty close to the testing instructions in #33046

**Comment by samueljseay** [comment]
@jonathansadowski Thanks!

**Review by psealock — APPROVED** [review]
I got [this to run successfully](https://github.com/psealock/woocommerce/actions/runs/3588931218) in my fork. Nice job here @samueljseay 🚢 

All the correct steps were undertaken

**Pull request merged**